plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/attribute-filter.block_theme.spec.ts

```diff
@@ -178,5 +271,95 @@ test.describe( 'Product Filter: Attribute Block', () => {
 
 			await expect( products ).toHaveCount( 1 );
 		} );
+
+		test( 'clear button appears after a filter is applied', async ( {
+			page,
+			dropdownBlockPost,
+		} ) => {
+			await page.goto( dropdownBlockPost.link );
+
+			const dropdownLocator = page.locator(
+				'.wc-interactivity-dropdown'
+			);
+
+			await expect( dropdownLocator ).toBeVisible();
+			await dropdownLocator.click();
+
+			const yellowOption = page.getByText( 'Yellow' );
+			await yellowOption.click();
+
+			// wait for navigation
+			await page.waitForURL( /.*filter_color=yellow.*/ );
+
+			const button = page.locator(
+				'.wp-block-woocommerce-product-filter-clear-button'
+			);
+
+			await expect( button ).toBeVisible();
+		} );
+
+		test( 'clear button hides after deselecting all filters', async ( {
+			page,
+			dropdownBlockPost,
+		} ) => {
+			await page.goto( dropdownBlockPost.link );
+
+			const dropdownLocator = page.locator(
+				'.wc-interactivity-dropdown'
+			);
+
+			await dropdownLocator.click();
+
+			const yellowOption = page.getByText( 'Yellow' );
+			await yellowOption.click();
+
+			// wait for navigation
+			await page.waitForURL( /.*filter_color=yellow.*/ );
+
+			await dropdownLocator.click();
+
+			const button = page.locator(
+				'.wp-block-woocommerce-product-filter-clear-button'
+			);
+
+			const removeFilter = page.locator(
+				'.wc-interactivity-dropdown__badge-remove'
+			);
+
+			await removeFilter.click();
+
+			await expect( button ).toBeHidden();
+		} );
+
+		test( 'filters are cleared after clear button is clicked', async ( {
+			page,
+			dropdownBlockPost,
+		} ) => {
+			await page.goto( dropdownBlockPost.link );
+
+			const dropdownLocator = page.locator(
+				'.wc-interactivity-dropdown'
+			);
+
+			await dropdownLocator.click();
+
+			const yellowOption = page.getByText( 'Yellow' );
+			await yellowOption.click();
+
+			// wait for navigation
+			await page.waitForURL( /.*filter_color=yellow.*/ );
+
+			const button = page
+				.locator( '.wp-block-woocommerce-product-filter-clear-button' )
+				.locator( 'button' );
+
+			await button.click();
+
+			const placeholder = await page
+				.locator( '.wc-interactivity-dropdown__placeholder' )
+				.textContent();
+
+			expect( placeholder ).toEqual( 'Select Color' );
```

Comment:
This could be rewritten using the built-in locators:

```suggestion
			const placeholder = page.getByText( 'Select Color' );

			await expect( placeholder ).toBeVisible();
```